tardis/adapters/sites/cloudstack.py

```diff
@@ -23,13 +23,14 @@
 
 class CloudStackAdapter(SiteAdapter):
     def __init__(self, machine_type: str, site_name: str):
-        self.configuration = getattr(Configuration(), site_name)
+        self._configuration = getattr(Configuration(), site_name)
         self.cloud_stack_client = CloudStack(
-            end_point=self.configuration.end_point,
-            api_key=self.configuration.api_key,
-            api_secret=self.configuration.api_secret,
+            end_point=self._configuration.end_point,
+            api_key=self._configuration.api_key,
+            api_secret=self._configuration.api_secret,
             event_loop=runtime._meta_runner.runners[asyncio].event_loop,
         )
+        self._machine_meta_data = self._configuration.MachineMetaData[machine_type]
         self._machine_type = machine_type
```

Comment:
Can you explain why you differentiate between ``_machine_meta_data`` and ``_machine_type`` when ``_machine_meta_data`` itself seems to be initiated with ``machine_type`` from configuration? What is the difference?

What is the intention to make this explicit while you seem to always be able to access the required value via ``_configuration``?

Further please consider that your ``SiteAdapter`` already implements a property for ``machine_meta_data`` accessing ``configuration`` to return the even same value.

Reply:
> Can you explain why you differentiate between `_machine_meta_data` and `_machine_type` when `_machine_meta_data` itself seems to be initiated with `machine_type` from configuration? What is the difference?

Machine types are just a list of supported "flavours" at a given site, it is a simple list of string.

```yaml
  MachineTypes:
    - Micro
    - Tiny
    - Medium
    - Large
    - Huge
    - Titan
```
This makes it easy to comment one flavour in the configuration in order to temporarily disable it.

The machine_meta_data contains information about number of cores, memory and disk available, where `machine_type` is a mapping node in yaml.

```yaml
MachineMetaData:
  Micro:
    Cores: 1
    Memory: 0.5
    Disk: 70
```

Reply:
> What is the intention to make this explicit while you seem to always be able to access the required value via `_configuration`?

Simplicity! The drone needs access to the `machine_meta_data` and has no direct access to it via the configuration. The drone does not now to which site or flavour it belongs. However, to set its `maximal_demand` it needs to know how many cores it has. 

https://github.com/MatterMiners/tardis/blob/7d1752e34487c3ee9274301f12c3778cfd512807/tardis/resources/drone.py#L68-L70

The drone gets this information via the `SiteAgent` which implements the `SiteAdapter` interface. Introducing a proporty will ensure that the information is present in all `SiteAdapter` implementations.

Reply:
> Further please consider that your `SiteAdapter` already implements a property for `machine_meta_data` accessing `configuration` to return the even same value.

The `SiteAdapter` should be the only place to implement this property except the `SiteAgent` which forwards the call to the corresponding site adapter.